### PR TITLE
 feat: Header validation with exact header names & header count match validation 

### PR DESCRIPTION
**Pull request by choudsc**
JIRA link (if applicable)

https://tools.hmcts.net/jira/browse/RDCC-1327

Change description

feat: Header validation with exact header names & header count match validation. 

Does this PR introduce a breaking change? (check one with "x")
No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/HeaderValidationProcessor.java [hunk 1]
```diff
@@ -60,36 +31,11 @@ public void process(Exchange exchange) throws Exception {
         String csv = exchange.getIn().getBody(String.class);
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/HeaderValidationProcessor.java:31]: Do we need to check null values for exchange ,getIn() and getBody() ?

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/util/HeaderUtil.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        jdbcTemplate.update(invalidHeaderSql, params);
+        TransactionStatus status = platformTransactionManager.getTransaction(def);
+        platformTransactionManager.commit(status);
+        throw new RouteFailedException(exceptionMsg + routeProperties.getFileName());
+    }
+
+    public String getInvalidJrdHeader(List<String> header, String binder) {
+        List<String> originalHeader;
+        String exceptionMsg = "";
+        Map<String, Object> headerMap = readYmlAsMap("header-mapping.yaml");
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/util/HeaderUtil.java:70]: can we remove hard code file name ?

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/util/HeaderUtil.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        sort(originalHeader);
+
+        if (originalHeader.size() == header.size() || header.size() > originalHeader.size()) {
+            List<String> invalidHeaderExtra = header.stream().sorted().filter(o -> (originalHeader.stream().sorted().filter(f -> f.equalsIgnoreCase(o)).count()) < 1).collect(Collectors.toList());
+            if (null != invalidHeaderExtra && !Collections.isEmpty(invalidHeaderExtra) && invalidHeaderExtra.toString().replaceAll("\\[\\]", "").length() > 0) {
+                exceptionMsg = "Invalid column(s) : " + invalidHeaderExtra.toString().replace(", ]", "]") + ". Please remove invalid column(s) from file.";
+            }
+        }
+        if (header.size() < originalHeader.size()) {
+            List<String> invalidHeaderLess = originalHeader.stream().sorted().filter(o -> (header.stream().sorted().filter(f -> f.equalsIgnoreCase(o)).count()) < 1).collect(Collectors.toList());
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/util/HeaderUtil.java:84]: Can this both If blocks be written in common ?

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/util/HeaderUtil.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.DefaultTransactionDefinition;
+
+import org.yaml.snakeyaml.Yaml;
+import uk.gov.hmcts.reform.juddata.camel.exception.RouteFailedException;
+import uk.gov.hmcts.reform.juddata.camel.route.beans.RouteProperties;
+
+@Service
+public class HeaderUtil {
+
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/util/HeaderUtil.java:35]: Please refactor. I think we can make more clean. Util is not required

**Pull request closed without merging**